src/VisualStudio/Core/Def/Implementation/ProjectSystem/Legacy/AbstractLegacyProject_ICompilerOptionsHostObject.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.Interop;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.Legacy
+{
+    internal partial class AbstractLegacyProject : ICompilerOptionsHostObject
+    {
+        private string _lastParsedCompilerOptions;
+
+        int ICompilerOptionsHostObject.SetCompilerOptions(string compilerOptions, out bool supported)
+        {
+            UpdateOptions(compilerOptions);
+            supported = true;
+            return VSConstants.S_OK;
+        }
+
+        protected abstract CommandLineArguments ParseCommandLineArguments(IEnumerable<string> splitArguments);
+
+        /// <summary>
+        /// Parses the given command line (if different from the last command line), and updates options using the latest state.
+        /// </summary>
+        /// <param name="commandLine">Optional command line to parse options. If null, default CommandLineArguments will be used.</param>
+        protected void UpdateOptions(string commandLine = null)
```

Comment:
This is shadowing the base method, which isn't what you want. This actually means there's a (possible) bug, as a call to `UpdateOptions` will toss the command line with the string.Empty. Just move the part here dealing with _lastParsedCompilerOptions in SetCompilerOptions above, and then this method goes away.